### PR TITLE
Validate selected compiler outputs on the CLI

### PR DESCRIPTION
**Pull request by cameel**
Implements part of #11629.
~Depends on #12070. In draft mode until that PR is merged.~ Merged.

This makes the compiler report errors when outputs that are unavailable in the current input mode are selected via CLI flags.

This was originally a part of a bigger PR that enables output selection in assembly mode but it actually also makes sense on its own and can be considered a separate feature/fix.

### CODE REVIEW

**Comment by cameel** [comment]
#12070 merged so this is now reviewable.

**Comment by cameel** [comment]
This change is technically breaking, but it could be argued that it's a bugfix (see https://github.com/ethereum/solidity/pull/11730#discussion_r706150636) if we consider missing CLI validations bugs rather than missing features.

File: solc/CommandLineParser.cpp [hunk 1]
```diff
@@ -460,6 +468,47 @@ bool CommandLineParser::parseLibraryOption(string const& _input)
...
+	static auto outputSupported = [](InputMode _mode, string_view _outputName)
+	{
+		static set<string> const compilerModeOutputs =
+			CompilerOutputs::componentMap() |
+			ranges::views::keys |
+			ranges::to<set>();
+
+		switch (_mode)
+		{
+		case InputMode::Compiler:
```
Comment (Marenz) [inline solc/CommandLineParser.cpp:482]: Ah, I thought we need explicit `[[fallthrough]];` but if this works, great!

**Review by Marenz — APPROVED** [review]

File: solc/CommandLineParser.cpp (hunk 1 above)
Comment (cameel) [reply solc/CommandLineParser.cpp:482]: We only need that if there's anything between the labels.

**Pull request merged**